Source/Core/InputCommon/ControllerInterface/ControllerInterface.cpp

```diff
@@ -319,6 +319,19 @@ bool ControllerInterface::AddDevice(std::shared_ptr<ciface::Core::Device> device
 
     NOTICE_LOG_FMT(CONTROLLERINTERFACE, "Added device: {}", device->GetQualifiedName());
     m_devices.emplace_back(std::move(device));
+
+    // We can't (and don't want) to control the order in which devices are added, but we
+    // need their order to be consistent, and we need the same one to always be the first, where
+    // present (the keyboard and mouse device usually). This is because when defaulting a
+    // controller profile, it will automatically select the first device in the list as its default.
+    std::sort(m_devices.begin(), m_devices.end(),
```

Comment:
`std::sort` is not guaranteed to preserve the order of elements that are equivalent. `std::stable_sort` should be used here.